ParticleObjects/particle_class.f90

```diff
@@ -111,6 +114,7 @@ module particle_class
     integer(shortInt)          :: fate = 0       ! Neutron's fate after being subjected to an operator
     integer(shortInt)          :: type           ! Particle type
     integer(shortInt)          :: collisionN = 0 ! Index of the number of collisions the particle went through
+    real(defReal)              :: collisionE = ZERO ! Collision energy
```

Comment:
I am not fully sure about putting the `collisionE` in the particle... 
For me the problem is that once the particle leaves a collision processor this information is missing necessary context: temperature and nuclide)